daemon/api_notices.go

```diff
@@ -232,6 +246,104 @@ func allowedNoticeTypesForInterface(iface string) []state.NoticeType {
 	return types
 }
 
+func postNotices(c *Command, r *http.Request, user *auth.UserState) Response {
+	requestUID, err := uidFromRequest(r)
+	if err != nil {
+		return Forbidden("cannot determine UID of request, so cannot create notice")
+	}
+
+	var payload struct {
+		Action string `json:"action"`
+		Type   string `json:"type"`
+		Key    string `json:"key"`
+		// NOTE: Data and RepeatAfter fields are not needed for snap-run-inhibit notices.
+	}
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return BadRequest("cannot decode request body: %v", err)
+	}
+
+	if payload.Action != "add" {
+		return BadRequest("invalid action %q", payload.Action)
+	}
+	if payload.Type != "snap-run-inhibit" {
+		return BadRequest(`invalid type %q (can only add "snap-run-inhibit" notices)`, payload.Type)
+	}
+	if len(payload.Key) > maxNoticeKeyLength {
+		return BadRequest("key must be %d bytes or less", maxNoticeKeyLength)
+	}
+
+	st := c.d.overlord.State()
+	st.Lock()
+	defer st.Unlock()
+
+	fromSnapCmd, err := requestFromSnapCmd(st, r)
+	if err != nil {
+		return InternalError("cannot check request source")
+	}
+	if !fromSnapCmd {
+		return Forbidden("only snap command can record notices")
+	}
+
+	exists, err := snapInstanceExists(st, payload.Key)
+	if err != nil {
+		return InternalError("cannot check snap in state: %v", err)
+	}
+	if !exists {
+		return BadRequest("snap %q does not exist", payload.Key)
+	}
+
+	noticeId, err := st.AddNotice(&requestUID, state.SnapRunInhibitNotice, payload.Key, nil)
+	if err != nil {
+		return InternalError("%v", err)
+	}
+
+	return SyncResponse(addedNotice{ID: noticeId})
+}
+
+func requestFromSnapCmd(st *state.State, r *http.Request) (bool, error) {
```

Comment:
This is a good check, thanks! I think it would be great as a general check for a new access checker like `snapClientAccess{}`.

Reply:
I think isRequestFrom would be a bit a clearer name for this, also it migh belong to access.go